Assets/Scripts/Model/Weapon.cs

```diff
@@ -1,29 +1,37 @@
 ﻿using UnityEngine;
 using System.Collections;
+using System.Collections.Generic;
 
 public class Weapon : MonoBehaviour, CanUpgrade
 {
     public float baseDamage;
-    public float baseRadius;
     public float baseRange;
     public float baseCooldown;
+	public float upgradeFactor;
+	public AudioClip shootSound;
 
-    // TODO change all these to private
-    public float currentDamage; 
-    public float currentRadius;
-    public float currentRange;
-    public float currentCooldown;
-    //
-
-    public float upgradeFactor;
+    private float currentDamage;
+    private float currentRange;
+    private float currentCooldown;
+	private List<CanReceiveDamage> targets;
```

Comment:
While I was writing a response I've realized that your solution might be even better than what I've thought initially xd nice work!